crates/bevy_render/src/lib.rs

```diff
@@ -236,3 +242,48 @@ fn check_for_render_resource_context(context: Option<Res<Box<dyn RenderResourceC
         );
     }
 }
+
+fn window_icon_changed(
```

Comment:
Something to consider with this system, is that it gets executed each frame. 
I'm curious if we can find some fun way to only have it execute when a window's icon is updated, or when the asset server has loaded it's value. 

Probably not, but it's something to think about 😅

Reply:
Hmm, yes. 
We could do a roundabout way via a command to bevy_winit and an event from there, but I dont think we should involve bevy_winit for that.